### PR TITLE
Upgrade Retail SDK Android

### PR DESCRIPTION
**Pull request by DanielOS7-TM**
Upgrade Retail SDK
Update reused `useIgnite` methods to useCallback's

### CODE REVIEW

**Comment by TakaGoto** [comment]
LGTM let me know if you're merging this before or after analytics listener work.

**Review by TakaGoto — APPROVED** [review]

**Comment by DanielOS7-TM** [comment]
@TakaGoto thanks for the review, I will wait till after the Android analytics changes is in before merging this 👍

**Comment by DanielOS7-TM** [comment]
Hey @TakaGoto we'll merge and release this today as a lot of project work needs this fix from the library

**Pull request merged**
